### PR TITLE
ICC: Implement some of Profile::from_pcs()

### PR DESCRIPTION
**Pull request by nico**
This implements conversion from profile connection space to the
device-dependent color for matrix-based profiles.

Inverting non-parametric curves is a TODO for now.

### CODE REVIEW

File: Userland/Libraries/LibGfx/ICC/Profile.cpp [hunk 1]
```diff
@@ -1471,6 +1472,134 @@ ErrorOr<FloatVector3> Profile::to_pcs(ReadonlyBytes color) const
...
+            // where redTRC^-1, greenTRC^-1, and blueTRC^-1 indicate the inverse functions of the redTRC greenTRC and
+            // blueTRC functions respectively.
+            // If the redTRC, greenTRC, or blueTRC function is not invertible the behaviour of the corresponding redTRC^-1,
+            // greenTRC^-1, and blueTRC^-1 function is undefined. If a one-dimensional curve is constant, the curve cannot be
+            // inverted."
+
+            // FIXME: Inverting matrix and curve on every call to this function is very inefficient.
+            auto const& redMatrixColumn = red_matrix_column();
+            auto const& greenMatrixColumn = green_matrix_column();
+            auto const& blueMatrixColumn = blue_matrix_column();
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGfx/ICC/Profile.cpp:1558]: `wrong_variable_name_case`

File: Userland/Libraries/LibGfx/ICC/TagTypes.h [hunk 2]
```diff
@@ -739,6 +739,45 @@ class ParametricCurveTagData : public TagData {
         VERIFY_NOT_REACHED();
     }
 
+    // y must be in [0..1].
+    float evaluate_inverse(float y) const
+    {
+        VERIFY(0.f <= y && y <= 1.f);
+
+        // See "Recommdations" section in https://www.color.org/whitepapers/ICC_White_Paper35-Use_of_the_parametricCurveType.pdf
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGfx/ICC/TagTypes.h:747]: "Recommendations"

A typo, in a Thakis PR?!?! Unthinkable

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
I have some nits!

I don't understand what the code actually does enough to say if it's any good but I trust you. :upside_down_face:

File: Userland/Libraries/LibGfx/ICC/Profile.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp:1558]: It matches the spec (https://color.org/specification/ICC.1-2022-05.pdf appendix, uh, F, I think was it) and elegantly sidesteps the variable name conflicting with the function name. Maybe fine?

**Comment by nico** [comment]
Fixed the typo, and also a grammaro in a commit message (it used to say `Make an "not yet implemented" string…`).

File: Userland/Libraries/LibGfx/ICC/Profile.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp:1558]: (it has the same name in `to_pcs()`, for the same reason)

Comment (GMTA) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp:1558]: As far as I know, we do not copy variables directly from specs. We do however try to adhere to the names. Something like `redMatrixColumn` in the spec would become `red_matrix_column` to adhere to our code style guidelines.

Comment (nico) [reply Userland/Libraries/LibGfx/ICC/Profile.cpp]: A'ight, done.

**Pull request merged**